### ISSUE TITLE
Suggestion: Destroy plans to show actual target object identities, not just TF naming

### ISSUE THREAD

**Issue by stevehorsfield** [hashicorp/terraform#10012]
### Terraform Version
0.6.16 (cannot upgrade because destroying based on existing TF code). No mention of improvement in CHANGELOG however.

### Affected Resource(s)
- All AWS resources, probably others as well

### Suggestion

I'm currently reviewing some destroy plans in Atlas for environments that share the same underlying AWS account. These environments differ only by variables and so the same logical names are present in the TF files but they create independent target objects, such as AWS IAM users, roles, S3 buckets, etc.

When creating the destroy plan, I get output like (names adjusted):

```
Path: /terraform/terraform.tfplan

- aws_iam_access_key.custom_user_access_key
- aws_iam_role.custom_role
- aws_iam_user.custom_user
- aws_s3_bucket.bucket
```

It would be ideal if we got output like:

```
Path: /terraform/terraform.tfplan

- aws_iam_access_key.custom_user_access_key (arn:....:user/john, access_key_id: abcsdsd)
- aws_iam_role.custom_role (arn:....:role/mkt)
- aws_iam_user.custom_user (arn:....:user/john)
- aws_s3_bucket.bucket (arn:...:bucket_john)
```

Currently, I have to examine the raw state to ensure that the affected objects are the intended ones.

This would be a good improvement for all plans, not just destroy plans.

**Comment by apparentlymart** [comment]
This is a nice idea, @stevehorsfield!

Given the limitations of Terraform's internal model I expect we wouldn't be able to get *exactly* the output you described, but we could do something where the value of the `id` attribute is shown; what exactly is stored in that attribute depends on the resource, but it's generally whatever we need to look up the object in the underlying API, which means it tends to be something humans can look up too.

We are already including this id in the log output for some operations, such as the "refreshing state" log lines, so seems like it should be pretty simple to include this same data in the plan output.

**Comment by stevehorsfield** [comment]
Sounds good to me, thanks @apparentlymart.

**Comment by RichardBronosky** [comment]
@stevehorsfield you say, "I have to examine the raw state to ensure that the affected objects are the intended ones". How is this done?

**Comment by stevehorsfield** [comment]
@RichardBronosky - We are using Terraform Enterprise. This provides a Web UI to look at the raw terraform state, i.e. the terraform.tfstate file. I then follow the identifier path, much like `terraform state show <full_path_name>` and find the appropriate AWS unique identifier, such as:

```
"id": "arn:aws:iam::<account_number>:policy/<policy_name>",
```

Then I know precisely which object will be affected.

**Comment by RichardBronosky** [comment]
Yes. The piece I was missing was `terraform state list` and `terraform state show`. Thanks!

**Comment by apparentlymart** [comment]
Hi all!

In the master branch we have a change to the plan rendering that includes, among other changes, a more complete rendering of destroy changes that is intended to give you the full context required to evaluate a destroy change, just as for any other change.

![](https://user-images.githubusercontent.com/20180/48109002-8ee13e80-e1fa-11e8-8cfc-66690c1155d7.png)

This will be included in the forthcoming v0.12.0 release. Since this is already merged, I'm going to close this out. Thanks for requesting this, and sorry for the long silence.
